examples/forms-signup.rs

```diff
@@ -329,17 +416,37 @@ fn fake_service(request: FakeApiRequest) {
             // Simulate this api taking a while
             thread::sleep(Duration::from_secs(1));
 
-            let mut errors = Map::new();
+            // Simulate some arbitrary server-side validation rules not-known to the form
+            fn has_more_than_four_digits(s: &str) -> bool {
+                s.chars().filter(|c| c.is_digit(10)).count() > 4
+            }
+
+            fn contains_year(s: &str) -> bool {
+                let re = Regex::new(r"(?:^|\D)(19[0-9]{2}|20[0-9]{2})(?:\D|$)").unwrap();
+                re.is_match(s)
+            }
+
+            let mut errors: Vec<u32> = Vec::default();
             if username == "admin" {
-                errors.insert(
-                    SignupField::Username,
-                    String::from("admin is a reserved username"),
+                errors.push(
+                    FakeApiSignupErrorCode::UsernameReserved.into(),
+                );
+            }
+            if username == "user" {
+                errors.push(
+                    FakeApiSignupErrorCode::UsernameUnavailable.into(),
```

Comment:
What does this extra if statement provide that the existing admin one didn't? Why are we complicating this example with extra arbitrary errors rather than keeping it as simple as possible to minimize confusion.